### ISSUE TITLE
Different Display Asset and Expenses vs. Revenue accounts 

### ISSUE THREAD

**Issue by GaryQ** [firefly-iii/firefly-iii#3050]
FF3 V 5.0.0 was present in Beta I also since I tested both. Can't confirm behavior on the demo site since it does not work in Edge

Period selected: 2019/01/01 2019/12/21
Expense account: (Asset account returns same result)

All transactions for account YadaYada between January 1, 2019 and December 31, 2019 
Returns as selectable months all 12 
![Snip1](https://user-images.githubusercontent.com/3269694/73067466-629cf680-3e6e-11ea-97dc-a17b13bb8ca2.jpg)

Revenue account does not display all 12 stops at Feb no January. Same result all revenue accounts

![Snip3](https://user-images.githubusercontent.com/3269694/73067627-d5a66d00-3e6e-11ea-93fc-340db2007a23.jpg)

**Comment by JC5** [comment]
I'm having a hard time replicating this but no matter.

What's the state of the revenue accounts? Any transactions in Dec '18, Jan '19 and Feb '19?

**Comment by GaryQ** [comment]
Sorry for the delay in responding. 
The TL;DR VERSION: It's definitely a reproducible bug and in all account types I have:
Asset, expense and revenue.

I did Figure out it's related to the periods collection. Traced back manually from the twig template to the ShowController. Seems a little bug has been hiding very quietly for a very long time as I know for a fact it goes back at least to 4.8.01 and it's still there in 5.0.1. Finding exactly what is causing it will be easier for you than for me. 

All transactions entered are strictly in January 2019. (Until I state I've moved forward this will always be my only month with data.) I decided to check my asset and expense accounts after closing in on what is causing the bug and it's the same scenario in all accounts. 

1. If the range selected is only January 2019 It shows the period box for Jan 2019 regardless of whether there was any transactions in the account

2. If any other range is selected. starting 2019-01-01 to whatever.... 
I  Tested 3 ranges: everything, 2019  and 2019-01-01 to 2019-02-28. Since they all gave the same result I stopped testing ranges.

3. The period box for Jan 2019 will display only if 1 of 2 specific conditions are met: No entry or an entry on Jan 1st.

4. If the account has a transaction anywhere in Jan 2019 but not one on the first the box will not display.

5. I went through all asset, revenue and a sampling of my expense accounts. The same behavior was observed.

6. I Took a revenue account with only one transaction and changed the date up and down and the only date that made the box appear was Jan 1st

7. I took a Credit card account that had no transactions and it displayed the box. Then entered a transaction on 2019-01-23 and the Jan 2019 period box was no longer displayed if a range other than only Jan 2019 was selected.

The charts and tables display the data regardless.

Conclusion: It's definitely a bug 
Personal opinion: That period box layout could be a future layout redesign. Cause it … 👎

**Comment by JC5** [comment]
Thanks, I'll see what I can find. 

> (Until I state I've moved forward this will always be my only month with data.)

I'm not going to remember this.

>  Personal opinion: That period box layout could be a future layout redesign. Cause it … 👎

I welcome your input.

**Comment by GaryQ** [comment]
I'm confident that you'll find it 👍

**Comment by JC5** [comment]
Let me know.

**Comment by GaryQ** [comment]
Sorry, A little quick to jump on the FIXED button there bud :)

It does display January 2019 now but....
Only Jan 2019 LOL I added a transaction in March 2019, 
Set range to everything and all I get is the Jan 2019 period box now
and yes flush cache and browser too plus browser in get from server so no caching to test.

**Comment by JC5** [comment]
I see it, working on it :)

**Comment by JC5** [comment]
Let me know what this code does on your end.

**Comment by GaryQ** [comment]
Thanks, Seems to be working fine now.
